tools/js-sdk-release-tools/src/changelog/extractMetaData.ts

```diff
@@ -69,8 +71,15 @@ export const extractExportAndGenerateChangelog = async (mdFilePathOld: string, m
 ) => {
     const metaDataOld = await readSourceAndExtractMetaData(mdFilePathOld);
     const metaDataNew = await readSourceAndExtractMetaData(mdFilePathNew);
-    const changeLog = changelogGenerator(metaDataOld, metaDataNew, oldSdkType, newSdkType);
-    logger.info('Generated changelog successfully:');
-    logger.info(changeLog.displayChangeLog());
-    return changeLog;
+    
+    try {
+        await mkdirp('./tmp-patch');
+        const astContext = await createAstContext(mdFilePathOld, mdFilePathNew, './tmp-patch');
+        const changeLog = changelogGenerator(metaDataOld, metaDataNew, oldSdkType, newSdkType, astContext);
```

Comment:
if an error throw out, will pipeline be able to resolve？I remember previously we have an issue in automation. wonder if we should add catch logic here?

Reply:
Yes, we already catch it here https://github.com/Azure/azure-sdk-tools/blob/main/tools/js-sdk-release-tools/src/autoGenerateInPipeline.ts#L98, still throw to let sdk automation know it's failed

Reply:
Yes, it's used here: https://github.com/Azure/azure-sdk-tools/blob/45e08850a0367b2a491e1ff57e9fffeff9e5ea08/tools/js-sdk-release-tools/src/llc/generateRLCInPipeline/generateRLCInPipeline.ts#L246
and catched and update the result here https://github.com/Azure/azure-sdk-tools/blob/main/tools/js-sdk-release-tools/src/llc/generateRLCInPipeline/generateRLCInPipeline.ts#L265
finally it will write the result to disk to let pipeline framework know the generation is failed: https://github.com/Azure/azure-sdk-tools/blob/main/tools/js-sdk-release-tools/src/autoGenerateInPipeline.ts#L110